app/src/main/java/com/github/palFinderTeam/palfinder/ui/login/LoginActivity.kt

```diff
@@ -173,20 +224,109 @@ class LoginActivity : AppCompatActivity() {
+            .addOnCompleteListener(this) { task ->
+                if (task.isSuccessful) {
+                    // Sign in success, update UI with the signed-in user's information
+                    Log.d(TAG, "createUserWithEmail:success")
+                    savePassword(email,password)
+                } else {
+                    // If user already in database, sign in
+                    if(task.exception is FirebaseAuthUserCollisionException){
+                        signIn(email, password)
+                    } else {
+                        //if creation fails, display error message
+                        signInSignUpFailure(task, "createUserWithEmail:failure")
+                    }
+                }
+            }
+        // [END create_user_with_email]
+    }
+
+    private fun signIn(email: String, password: String) {
+        // [START sign_in_with_email]
+        auth.signInWithEmailAndPassword(email, password)
+            .addOnCompleteListener(this) { task ->
+                if (task.isSuccessful) {
+                    // Sign in success, update UI with the signed-in user's information
+                    Log.d(TAG, "signInWithEmail:success")
+                    savePassword(email,password)
+                } else {
+                    // If sign in fails, display a message to the user.
+                    signInSignUpFailure(task, "signInWithEmail:failure")
+                }
+            }
+        // [END sign_in_with_email]
+    }
+
+    private fun signInSignUpFailure(task: Task<AuthResult>, logText: String) {
+        Log.w(TAG, logText, task.exception)
+        Toast.makeText(
+            baseContext, "Authentication failed.",
+            Toast.LENGTH_SHORT
+        ).show()
+        updateUI(null)
+    }
+
+    private fun savePassword(email: String, password: String) {
+        val signInPassword = SignInPassword(email, password)
+        val savePasswordRequest =
+            SavePasswordRequest.builder().setSignInPassword(signInPassword).build()
+        Identity.getCredentialSavingClient(this).savePassword(savePasswordRequest)
+            .addOnSuccessListener { result ->
+                try {
+                startIntentSenderForResult(
+                    result.pendingIntent.intentSender,
+                    REQUEST_CODE_GIS_SAVE_PASSWORD,  /* fillInIntent= */
+                    null,  /* flagsMask= */
+                    0,  /* flagsValue= */
+                    0,  /* extraFlags= */
+                    0,  /* options= */
+                    null
+                )}catch (e: IntentSender.SendIntentException) {
+                    Log.e(TAG, "Couldn't save password: ${e.localizedMessage}")
+                }
+            }
+            .addOnFailureListener(this) { e ->
+                // No saved credentials found. Launch the One Tap sign-up flow, or
+                // do nothing and continue presenting the signed-out UI.
+                Log.d(TAG, e.localizedMessage)
```

Comment:
If the user refuse to save credentials and tries to log in, it will go through this branch and the UI will not update